File: quiz-backend/src/main/java/pl/sdacademy/projektplus/quiz/dto/GameStateDto.java
package pl.sdacademy.projektplus.quiz.dto;

import lombok.Data;
import pl.sdacademy.projektplus.quiz.frontend.GameOptions;

import java.util.List;

@Data
public class GameStateDto {
    private final GameOptions gameOptions;
    private final int currentQuestionNumber;
    private final int points;

    private final String currentQuestionText;
    private final List<String> currentQuestionAnswers;
}

File: quiz-frontend/src/app/dtos/gameState.dto.ts
import {GameOptionsDto} from './gameOptions.dto';

export class GameStateDto {
    currentQuestionNumber: number;
    points: number;
    gameOptions: GameOptionsDto;
    currentQuestionText: string;
    currentQuestionAnswers: string[];

    constructor(currentQuestionNumber: number, points: number, gameOptions: GameOptionsDto, currentQuestionText: string, currentQuestionAnswers: string[]) {
        this.currentQuestionNumber = currentQuestionNumber;
        this.points = points;
        this.gameOptions = gameOptions;
        this.currentQuestionText = currentQuestionText;
        this.currentQuestionAnswers = currentQuestionAnswers;
    }
}
